### ISSUE TITLE
TRT inference results are different from ONNX inference results

### ISSUE THREAD

**Issue by yo1990** [NVIDIA/TensorRT#2019]
## Description

The inference results of the model converted from ONNX with trtexec are clearly different from the inference results of the original ONNX.

## Environment

**TensorRT Version**:  8.0.1
**NVIDIA GPU**:  NVIDIA GeForce RTX 3070
**NVIDIA Driver Version**:  470.103.01
**CUDA Version**: 11.4
**CUDNN Version**: 
**Operating System**: Ubuntu 18.04
**Python Version (if applicable)**: 3.7
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

## Relevant Files

osnet_ms_d_c.onnx
https://drive.google.com/file/d/1jAsfjale3GvfcCQL3JOT9LBqJGxWp4Uw/view?usp=sharing

## Steps To Reproduce

1. ONNX -> TRT convert
```
/usr/src/tensorrt/bin/trtexec --onnx=osnet_ms_d_c.onnx --saveEngine=osnet_ms_d_c.onnx.engine
```
2. Make test input
```
import numpy as np

input_batch = np.random.randn(1, 3, 256, 128).astype(np.float32)
np.save("input.npy", input_batch)
```
3. Show ONNX inference results
```
import numpy as np
import onnxruntime as rt

input_batch = np.load("input.npy")
sess = rt.InferenceSession("osnet_ms_d_c.onnx")
output = sess.run(["1049"], {"input.1": input_batch})[0]
print(output.reshape(-1).tolist()[:20])

# output : [0.0, 0.028274890035390854, 0.011932285502552986, 0.0, 0.002865843242034316, 0.0, 0.06866644322872162, 0.017567593604326248, 0.0, 0.05154433846473694, 0.04393055662512779, 0.0, 0.0, 0.0027602179907262325, 0.0, 0.023882480338215828, 0.01876200921833515, 0.0, 0.03771236911416054, 0.0]
```
4. Show TRT inference results
```
import numpy as np
import pycuda
import pycuda.driver as cuda
import pycuda.autoinit
import tensorrt as trt

input_batch = np.load("input.npy")
with trt.Runtime(trt.Logger(trt.Logger.WARNING)) as runtime:
    with open("osnet_ms_d_c.onnx.engine", "rb") as f:
        engine = runtime.deserialize_cuda_engine(f.read())
    with engine.create_execution_context() as context:
        stream = cuda.Stream()
        input_binding = cuda.mem_alloc(input_batch.nbytes)
        cuda.memcpy_htod_async(input_binding, input_batch, stream)
        # context.set_binding_shape(0, input_batch.shape)

        output = cuda.pagelocked_empty(trt.volume((1, 3, 256, 128)), dtype=np.float32)
        output_binding = cuda.mem_alloc(output.nbytes)
        context.execute_async_v2(bindings=(input_binding, output_binding), stream_handle=stream.handle)

        cuda.memcpy_dtoh_async(output, output_binding, stream)
        stream.synchronize()
        print(output.reshape(-1).tolist()[:20])

# output : [0.0, 0.45159661769866943, 0.19070151448249817, 0.0, 0.04596791788935661, 0.0, 1.1008120775222778, 0.28185680508613586, 0.0, 0.8254732489585876, 0.7035995721817017, 0.0, 0.0, 0.04433565214276314, 0.0, 0.38196754455566406, 0.2996152341365814, 0.0, 0.6039710640907288, 0.0]
```

The inference results are clearly different, not an error.
Is it a problem with the procedure?
Or is the model conversion failing?
Please help...

**Comment by nvpohanh** [comment]
@yo1990 Could you try TRT 8.2 or 8.4EA to see if it is fixed in later version?

Based on the output values you shared, it seems that TRT's output values happen to be 16x of ONNXRT's output values. That means TRT may not apply the final normalization correctly.

**Comment by yo1990** [comment]
@nvpohanh 
Thanks for your reply.
I tried TRT 8.2.5.1 and 8.4.0.6 respectively, no change.
Is there any other possible cause?

**Comment by nvpohanh** [comment]
Thanks for trying. We will try repro this issue and keep you posted.

**Comment by nvpohanh** [comment]
I can repro this locally. I have filed an internal tracker (id: 3666457). Our engineer will debug and fix this issue. Will let you know if I get any updates. Thanks for reporting this!

**Comment by yo1990** [comment]
@nvpohanh 
Hi.
Any updates on this issue?

**Comment by nvpohanh** [comment]
we are still actively investigating this

**Comment by nvpohanh** [comment]
This bug has been fixed internally and will be available in TRT 8.5 release. Thanks for reporting the issue

**Comment by ttyio** [comment]
Closing since the 8.5 is released, thanks!
